icu4c/source/test/fuzzer/ubidi_fuzzer.cpp

```diff
@@ -0,0 +1,181 @@
+// © 2024 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// Fuzzer for ICU bidi.
+
+#include <cstring>
+
+#include "fuzzer_utils.h"
+
+#include "unicode/ubidi.h"
+
+#define MAXLEN 200
+
+static void sub2(UBiDi *bidi) {
+  UErrorCode errorCode = U_ZERO_ERROR;
+  int32_t i;
+  int32_t runCount;
+  ubidi_getDirection(bidi);
+  ubidi_getParaLevel(bidi);
+  ubidi_getReorderingMode(bidi);
+  ubidi_getReorderingOptions(bidi);
+  runCount = ubidi_countRuns(bidi, &errorCode);
+  if (U_FAILURE(errorCode)) {
+  } else {
+    for (i = 0; i < runCount; i++) {
+      int32_t start, len;
+      ubidi_getVisualRun(bidi, i, &start, &len);
+    }
+  }
+}
+
+static void matchingPair(UBiDi *bidi, int32_t i) {
+  UBiDiLevel level;
+  ubidi_getLogicalRun(bidi, i, NULL, &level);
+}
+
+static bool sub1(UBiDi *bidi) {
+  int32_t i, idx, logLimit, visLimit;
+  bool testOK, errMap, errDst;
+  UErrorCode errorCode = U_ZERO_ERROR;
+  int32_t visMap[MAXLEN];
+  int32_t logMap[MAXLEN];
+  ubidi_getVisualMap(bidi, visMap, &errorCode);
+  ubidi_getLogicalMap(bidi, logMap, &errorCode);
+  if (U_FAILURE(errorCode)) {
+    return false;
+  }
+
+  testOK = true;
+  errMap = errDst = false;
+  logLimit = ubidi_getProcessedLength(bidi);
+  visLimit = ubidi_getResultLength(bidi);
+  for (i = 0; i < logLimit; i++) {
+    idx = ubidi_getVisualIndex(bidi, i, &errorCode);
+    if (idx != logMap[i]) {
+      errMap = true;
+    }
+    if (idx == UBIDI_MAP_NOWHERE) {
+      continue;
+    }
+    if (idx >= visLimit) {
+      continue;
+    }
+    matchingPair(bidi, i);
+  }
+
+  if (U_FAILURE(errorCode)) {
+    return false;
+  }
+  if (errMap) {
+    if (testOK) {
+      sub2(bidi);
+      testOK = false;
+    }
+    for (i = 0; i < logLimit; i++) {
+      ubidi_getVisualIndex(bidi, i, &errorCode);
+    }
```

Comment:
What's this loop doing?  Are you just calling `ubidi_getVisualIndex()` again to see if it crashes or misbehaves?